wordpress-mobile/WordPress-iOS
# Xpost Suggestions for Gutenberg

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/2602
Related PRs:
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/2746
- https://github.com/WordPress/gutenberg/pull/26211
- https://github.com/wordpress-mobile/WordPress-Android/pull/13184

This PR adds to Gutenberg the ability to xpost to another site.

### To test

**What you'll need**: A WordPress.com site that is capable of xposting to one or more other sites
- Make sure some of the destination sites have a blavatar photo (site photo) set
- The ability to verify that the destination sites were successfully xpost'ed to

_Note: https://wordpress.com/p2/ can be used to create xpost-capable sites._

**Xpost in Gutenberg**

- Open the block editor
- In a Paragraph or any other rich text block, type <code>+</code> and verify that the list of suggestions is shown
- Make sure that the blavatar images load for sites which have a blavatar image
- Select one of the suggestions to add it to the post
- Expect the site name (not its URL) to be added to the post
- Publish the post
- Verify that the destination site received the xpost

Please note: There are improvements coming to the Xpost feature before it will be shipped, this PR aims to bring Xposts the same level of functionality Mentions currently has, nothing more, nothing less. (Improvements include a request throttle, prefetching, analytics, etc.)

### PR submission checklist
